tiledb/sm/array_schema/dimension.cc

```diff
@@ -1591,209 +1721,24 @@ std::string Dimension::tile_extent_str() const {
   return apply_with_type(g, type_);
 }
 
-void Dimension::set_crop_range_func() {
-  auto g = [&](auto T) {
-    if constexpr (tiledb::type::TileDBNumeric<decltype(T)>) {
-      crop_range_func_ = crop_range<decltype(T)>;
-    }
-  };
-  apply_with_type(g, type_);
-}
-
-void Dimension::set_domain_range_func() {
-  auto g = [&](auto T) {
-    if constexpr (tiledb::type::TileDBFundamental<decltype(T)>) {
-      domain_range_func_ = domain_range<decltype(T)>;
-    }
-  };
-  apply_with_type(g, type_);
-}
-
-void Dimension::set_ceil_to_tile_func() {
-  auto g = [&](auto T) {
-    if constexpr (tiledb::type::TileDBFundamental<decltype(T)>) {
-      ceil_to_tile_func_ = ceil_to_tile<decltype(T)>;
-    }
-  };
-  apply_with_type(g, type_);
-}
-
-void Dimension::set_coincides_with_tiles_func() {
-  auto g = [&](auto T) {
-    if constexpr (tiledb::type::TileDBFundamental<decltype(T)>) {
-      coincides_with_tiles_func_ = coincides_with_tiles<decltype(T)>;
-    }
-  };
-  apply_with_type(g, type_);
-}
-
-void Dimension::set_compute_mbr_func() {
-  if (!var_size()) {  // Fixed-sized
-    compute_mbr_var_func_ = nullptr;
-    auto g = [&](auto T) {
-      if constexpr (tiledb::type::TileDBNumeric<decltype(T)>) {
-        compute_mbr_func_ = compute_mbr<decltype(T)>;
-      }
-    };
-    apply_with_type(g, type_);
-  } else {  // Var-sized
+void Dimension::set_dimension_dispatch() {
+  if (var_size()) {
+    dispatch_ =
+        tdb_unique_ptr<DimensionDispatch>(tdb_new(DimensionVarSize, *this));
     assert(type_ == Datatype::STRING_ASCII);
-    compute_mbr_func_ = nullptr;
-    compute_mbr_var_func_ = compute_mbr_var<char>;
+  } else  // Fixed-sized
+  {
```

Comment:
Some weird indentation here. Might need to manually move that brace between `else` and `// Fixed-size`